gems/pending/disk/modules/miq_disk_cache.rb

```diff
@@ -0,0 +1,109 @@
+require 'rufus/lru'
+require_relative "../MiqDisk"
+require 'ostruct'
+
+module MiqDiskCache
+  MIN_SECTORS_TO_CACHE = 32
+  DEF_CACHE_SIZE       = 100
+  DEBUG_CACHE_STATS    = false
+
+  attr_reader :d_size, :blockSize, :cache_size, :sectors_to_cache, :downstream
+
+  def self.new(downstream, d_info = nil)
```

Comment:
I would leave it the way it is. Going forward, I want to convert all of the current modules into classes, and doing it this way will minimize client changes when we do so. The idea is to be able to construct a delegation chain IO stack, but that will require more changes to the MiqDisk class. For now, this gives us what we need for Azure, and gets us part way to where we want to be.